### ISSUE TITLE
Decrypting PGP with Private Key and no password

### ISSUE THREAD

**Issue by PrzemyslawKlys** [mattosaurus/PgpCore#297]
Hi,

So I've a PowerShell project https://github.com/EvotecIT/PSPGP which uses your library to play with PGP.

One of the features which seemed working before was with version 5.13.0 to decrypt PGP without a password. 

https://github.com/EvotecIT/PSPGP/blob/6ec937ea03869b2a0838ecb2f101e50d8391edaa/Public/Unprotect-PGP.ps1#L54-L58

Basically I was able to provide empty password and it would work. Now when Itry to upgrade module to 6.5.0 version it throws an error:`WARNING: Unprotect-PGP - Can't decrypt files because: Exception calling ".ctor" with "1" argument(s): "unknown object in stream 47"` when I am using `PgpCore.EncryptionKeys new(string privateKey, string passPhrase)` with empty password. 

Did something change? Or maybe it never worked and I imagined it?

**Comment by mattosaurus** [comment]
Hi, this did previously work but you're right that it no longer seems to. Thanks for letting me know.

I'll hopefully get a chance to look into this later this week.

**Comment by mattosaurus** [comment]
I think this may be to do with creating an instance of `EncryptionKeys` with a single argument.

```
var privateKeyFileInfo = new FilInfo(pathToPrivateKeyFile");
var decryptionKeys = new EncryptionKeys(privateKeyFileInfo);
```

If you do this then it assumes it's a public key rather than a private key and returns the rather unhelpful error you were recieveing.

If you pass en empty string through as the password then it will recognise it as a private key as expected.

```
var privateKeyFileInfo = new FilInfo(pathToPrivateKeyFile");
var decryptionKeys = new EncryptionKeys(privateKeyFileInfo, String.Empty);
```

I've improved the error message for this and added a couple of tests to demonstrate it working.

https://github.com/mattosaurus/PgpCore/pull/298

Let me know if this helps

**Comment by PrzemyslawKlys** [comment]
I'll give it a try, thank you.

**Comment by PrzemyslawKlys** [comment]
I always had:

![image](https://github.com/user-attachments/assets/51625c2f-e0a0-4f94-996a-b6411995c9cc)

So it was always providing empty string in form of "". But regardless if I do this, or provide [string]::Empty I always get unknown object in stream34.

After further debugging it seems it was related to `New-PGPKey -FilePathPublic $PSScriptRoot\Keys\PublicPGP3.asc -FilePathPrivate $PSScriptRoot\Keys\PrivatePGP3.asc -Strength 4096 -Certainty 8` the way I created public key for this specific test. When I removed strength and certainty it works. 

So I guess if the key is created using:

![image](https://github.com/user-attachments/assets/669c5fe1-68a4-4125-a08a-cd2a876ff9f4)

It's breaking things, I guess it may require other parameters to be set as well.
